### PR TITLE
Django 4.2 support

### PR DESCRIPTION
**Pull request by salman2013**
1. Add Django 4.2 support
2. Bump the version to '2.4.0'

Ticket ref: https://github.com/edx/upgrades/issues/139

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -119,6 +119,7 @@ def is_requirement(line):
         'Framework :: Django',
         'Framework :: Django :: 3.2',
         'Framework :: Django :: 4.0',
```
Comment (nedbat) [inline setup.py:121]: We probably want to remove 4.0 from the list.

File: tox.ini [hunk 2]
```diff
@@ -19,6 +19,7 @@ setenv =
     DJANGO_SETTINGS_MODULE = test_settings
 deps = 
     django32: Django>=3.2,<4.0
+    django42: Django>=4.2,<4.3
     django40: Django>=4.0,<4.1
```
Comment (nedbat) [inline tox.ini:23]: ditto here about removing 4.0

File: tox.ini [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py38-django{32,40}-drf{latest},quality,docs
+envlist = py38-django{32, 42}-drf{latest},quality,docs
```
Comment (nedbat) [inline tox.ini:2]: Is that extra space ok?

Comment (salman2013) [reply tox.ini]: Remove it.

File: setup.py (hunk 1 above)
Comment (salman2013) [reply setup.py]: I have removed this.

File: tox.ini (hunk 2 above)
Comment (salman2013) [reply tox.ini]: I have removed this.

**Review by nedbat — APPROVED** [review]

**Pull request merged**